### PR TITLE
Add meaningfull error messages for registration form

### PR DESCRIPTION
**Pull request by danialmalik**
#### Story Link
https://edlyio.atlassian.net/browse/EDE-379

#### PR Description
image

Add meaningful error messages for registration form

### CODE REVIEW

File: openedx/features/ucsd_features/additional_registration_fields.py [hunk 1]
```diff
@@ -13,12 +13,27 @@ class AdditionalRegistrationFieldsForm(ModelForm):
...
+                'required': 'Please select your Ethnicity.',
+            },
+            'age': {
+                'required': 'Please select your Age Group.',
+            },
+            'education': {
+                'required': 'Please select your Eduction.',
+            },
+            'howheard': {
+                'required': 'Please tell us How did you hear about us.'
```
Comment (HamzaIbnFarooq) [inline openedx/features/ucsd_features/additional_registration_fields.py:34]:
```suggestion
                'required': 'Please tell, how did you hear about us.'
```
sounds good?

File: openedx/features/ucsd_features/additional_registration_fields.py [hunk 2]
```diff
@@ -13,12 +13,27 @@ class AdditionalRegistrationFieldsForm(ModelForm):
...
+                'required': 'Please select your Gender.',
+            },
+            'ethnicity': {
+                'required': 'Please select your Ethnicity.',
+            },
+            'age': {
+                'required': 'Please select your Age Group.',
+            },
+            'education': {
+                'required': 'Please select your Eduction.',
```
Comment (HamzaIbnFarooq) [inline openedx/features/ucsd_features/additional_registration_fields.py:31]:
```suggestion
                'required': 'Please select your Education.',
```

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 1 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py:34]: I don't think this makes much of a difference.

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 2 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py:31]: done

Comment (HamzaIbnFarooq) [reply openedx/features/ucsd_features/additional_registration_fields.py]: what if we use 'Education Level' here?

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 1 above)
Comment (HamzaIbnFarooq) [reply openedx/features/ucsd_features/additional_registration_fields.py:34]: Firstly the 'H' of how needs to be small, secondly jo can use the sentence as you formed if was just trying to ignore two 'us' from the sentence, it's just an opinion

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 2 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py]: done

**Comment by edtecheco** [comment]
Your PR has finished running tests. There were no failures.

File: openedx/features/ucsd_features/additional_registration_fields.py [hunk 3]
```diff
@@ -13,12 +13,27 @@ class AdditionalRegistrationFieldsForm(ModelForm):
     """
     The fields on this form are derived from the AdditionalRegistrationFields model in models.py.
     """
-    def __init__(self, *args, **kwargs):
-        super(AdditionalRegistrationFieldsForm, self).__init__(*args, **kwargs)
 
     class Meta(object):
         model = AdditionalRegistrationFields
         fields = ('gender_nb', 'ethnicity', 'age', 'education', 'howheard')
+        error_messages = {
```
Comment (mumarkhan999) [inline openedx/features/ucsd_features/additional_registration_fields.py:20]: You can also use help text with error messages here like given below
```
class Meta(object):
        error_messages = {
             ....
        }
        help_texts = {
            'registration_number': 'Enter your registration number provided from your school.',
            'first_name': 'Enter your first name.',
            'surname': 'Enter your surname.',
            'other_name' : 'Enter your nick name.',
            'current_level': 'Select your current level in school during admission.',
            'mobile_number': 'Enter your mobile number.',
            'department': 'Select your department',
            'programme': 'Select the programme that you have been enrolled in.',
        }

```

Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py:20]: I don't think these are required as all of the custom fields are dropdowns. So there are limited, self-descriptive options there. Also, the message will be hidden behind the dropdown regardless.
image

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 1 above)
Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py:34]: updated

File: openedx/features/ucsd_features/additional_registration_fields.py (hunk 3 above)
Comment (mumarkhan999) [reply openedx/features/ucsd_features/additional_registration_fields.py:20]: image
No, the text didn't hide behind the dropdown. When you click on the drop-down menu for the first time it shows the help text

Comment (danialmalik) [reply openedx/features/ucsd_features/additional_registration_fields.py:20]: I've added the help texts/

**Comment by edtecheco** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by danialmalik** [comment]
jenkins run python

**Pull request merged**